webapp/src/components/FilterPanel.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from "react";
+
+interface FilterTagProps {
+    tag: string;
+    selected: boolean;
+    onClick: (selected: string) => void;
+}
+
+export class FilterTag extends React.Component<FilterTagProps> {
+    constructor(props: FilterTagProps) {
+        super(props)
+        this.clickHandler = this.clickHandler.bind(this);
+    }
+
+    render() {
+        return <div className="filter-tag">
+            <div className="filter-tag-box" role="checkbox" onClick={this.clickHandler} aria-checked={this.props.selected}>
+                <i className={this.props.selected ? `icon check square outline` : `icon square outline`}></i>
+            </div>
+            <div className="filter-tag-name" role="button" onClick={this.clickHandler}>{lf(this.props.tag)}</div>
+        </div>
+    }
+
+    protected clickHandler() {
+        this.props.onClick(this.props.tag);
+    }
+}
+
+
+interface FilterPanelSubheadingProps {
+    subheading: string;
+    buttonText?: string;
+    buttonAction?: () => void;
+}
+
+export class FilterPanelSubheading extends React.Component<FilterPanelSubheadingProps> {
+
+    render() {
+        return <div className="filter-subheading-row">
+            <div className="filter-subheading-title">{`${lf(this.props.subheading)}:`}</div>
+            {this.props.buttonText && <div className="filter-subheading-button" role="button" onClick={this.props.buttonAction}>{lf(this.props.buttonText)}</div>}
```

Comment:
can't use lf like this, parameter to lf has to be a string literal. I'm surprised nothing in the build is yelling here.

(solution is either `lf` when passing as a prop, which is preferable, or `rlf` here and making sure any possible text is either extracted from definitions during build or in a hacky commented out `lf` like https://github.com/microsoft/pxt/blob/master/webapp/src/container.tsx#L25)

```suggestion
            {this.props.buttonText && <div className="filter-subheading-button" role="button" onClick={this.props.buttonAction}>{rlf(this.props.buttonText)}</div>}
```

Reply:
ahhhhhhhhhhhhhhhhh okay so for tags, should I rlf, and add comments in the top the hacky way, and then we just gotta remember to add it here if we add more tags?

Reply:
For now that would work, we should probably be pulling out the tags in the build and uploading them to crowdin if we're displaying them to the user but that's more work / not something that feels necessary for this change.

Reply:
ooo yeah makes sense makes sense